app/src/components/mapSearch.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect, FunctionComponent } from "react";
+import {APIProvider, Marker, AdvancedMarker, Map} from '@vis.gl/react-google-maps';
+import GooglePlacesAutocomplete, { geocodeByPlaceId, getLatLng} from 'react-google-places-autocomplete';
+import TextForm from "./textForm";
+
+export default function MapSearch() {
+    const [map, setMap] = useState({
+        lat: 48.4284,
+        lng: -123.3656,
+    });
+    const [radius, setRadius] = useState(10);
+    const [search_value, setSearchValue] = useState(null);
+
+    return (
+        <div className="mt-3 max-w-40">
+            <GooglePlacesAutocomplete //package for the google places API autocomplete search bar
+            apiKey="API-KEY-HERE"
```

Comment:
Shouldn't the API key be imported from the .env file?

Reply:
Tried to get it to work but couldn't, doesn't github pages use a different implementation with secret pages? I figured I was wasting my time if GitHub was just going to do it differently.